### PR TITLE
npm-cache-install script that can work on any project

### PR DESCRIPTION
**Pull request by antialias**

### CODE REVIEW

**Comment by antialias** [comment]
this will require another major version bump

File: src/common.sh [hunk 1]
```diff
@@ -15,15 +15,25 @@ then
 	echo cacheInstallFromPackageJson $cacheInstallFromPackageJson
 	# let (package.json).cacheInstall override values in .npm-cache-install
 	export cacheInstallHost=$(node -e "console.log(($packageJson).cacheInstall.host || '')")
-	export cacheInstallDest=$(node -e "console.log(($packageJson).cacheInstall.path || '')")
+	export cacheInstallPath=$(node -e "console.log(($packageJson).cacheInstall.path || '')")
+fi
+if [[ -z "$cacheInstallHost" ]]
+then
```
Comment (cheryly279) [inline src/common.sh]: It doesn't look like we are actually pulling the values from .npm-cache-install? I don't think we should just hard code them here. What if someone wants to override for their system globally?

Comment (antialias) [inline src/common.sh]: > Or are those values pulled in via line 10 above?

Yes, that is what is happening

**Comment by cheryly279** [comment]
@antialias Looks good to me. I didn't get a chance to test - can you ask sysops to set it up so that your tests run as part of the PR? I also don't know much about Travis CI so I'll defer to you and merge this.

**Comment by antialias** [comment]
I've been bugging ross about this since yesterday.

**Comment by cheryly279** [comment]
I asked sysops yesterday about a different repo and they did it lickety-split. Just go to them.

Also it's not pressing, but do we want to fall back completely to values like localhost, or do we want to error out with a message so someone knows something is wrong with their setup?

**Comment by antialias** [comment]
I think only ross has access to the travis dashboard.

Defaulting to localhost is fine. This will let users cache locally, which is still helpful.

**Comment by antialias** [comment]
published as 4.0.0

**Pull request merged**